rotkehlchen/inquirer.py

```diff
@@ -1420,15 +2497,131 @@ def _query_fiat_pair(
+                    log.error("Yahoo Finance oracle not available for stock refresh")
+                    return
+                    
+                success_count = 0
+                for idx, asset in enumerate(assets_to_refresh):
+                    try:
+                        custom_asset = asset.resolve()
+                        if not (isinstance(custom_asset, CustomAsset) and 
+                                hasattr(custom_asset, 'custom_asset_type') and 
+                                custom_asset.custom_asset_type.lower() in ('stock', 'etf')):
+                            log.debug(
+                                f"Skipping non-stock/ETF asset in refresh",
+                                asset=asset.identifier
+                            )
+                            continue
+                            
+                        log.debug(
+                            f"Refreshing stock/ETF price ({idx+1}/{len(assets_to_refresh)})",
+                            asset=asset.identifier,
+                            name=getattr(custom_asset, 'name', 'unknown')
+                        )
+                        
+                        price = yahoo.query_custom_asset_price(
+                            custom_asset=custom_asset,
+                            to_currency='USD'
+                        )
+                        
+                        if price != ZERO_PRICE:
+                            log.info(
+                                f'Successfully refreshed price',
+                                asset=asset.identifier,
+                                name=getattr(custom_asset, 'name', 'unknown'),
+                                price=price
+                            )
+                            # Store the updated price
+                            Inquirer._store_custom_asset_price(asset, price)
+                            success_count += 1
+                        else:
+                            log.warning(
+                                f'Yahoo Finance returned zero price for {asset.identifier}'
+                            )
+                            
+                        # Add a small delay to avoid rate limiting
+                        time.sleep(0.5)
+                        
+                    except Exception as e:
+                        log.error(
+                            f'Failed to refresh stock/ETF price',
+                            asset=getattr(asset, 'identifier', 'unknown'),
+                            error=str(e)
+                        )
+                        
+                log.info(f"Completed background refresh of stock/ETF prices. Updated {success_count}/{len(assets_to_refresh)} prices")
+                
+            except Exception as e:
+                log.error(f"Background stock/ETF refresh process failed: {e}")
+        
+        # Start the background thread
+        thread = threading.Thread(target=refresh_all_stocks_bg)
```

Comment:
no no no don't do that. Do not use threading in rotki. Anyway as i said all of these changes need to go away